### DISCUSSION TITLE
morph replace on nullable morph

### DISCUSSION THREAD

**Discussion by falur** [spatie/laravel-medialibrary#1946] [Ideas] [answered]
if I need to upload the file first and then attach it to the model in handler. I can’t do it now.
The solution that I see

```php
$table->morphs('model');
// replace on
$table->nullableMorphs('model');
```

And add the method 
```php
public function addExistingMedia($id) 
{
// Logic
}
```

**Comment by freekmurze** [comment, marked as answer]
This is something we'll solve in the upcoming https://medialibrary.pro
